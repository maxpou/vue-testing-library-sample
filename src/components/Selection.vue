<template>
  <div>
    <div class="ui container" v-if="mySelection.length > 0">
      <h2 class="sectionTitle">My selection</h2>
      <div class="ui relaxed divided list">
        <div class="item" v-for="item in mySelection">
          <div class="right floated content">
            <div class="ui button" @click="removeFromSelection(item)">-</div>
          </div>
          <i class="large rocket middle aligned icon"></i>
          <div class="content">
            <a class="header">{{ item.name }}</a>
            <div class="left meta">{{ item.manufacturer }}</div>
            <div class="left description">{{ item.cost_in_credits }}</div>
          </div>
        </div>
      </div>
    </div>

  </div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex'

export default {
  computed: mapGetters({
    mySelection: 'mySelection'
  }),
  methods: mapActions({
    removeFromSelection: 'removeFromSelection'
  })
}
</script>

<style scoped>
</style>
