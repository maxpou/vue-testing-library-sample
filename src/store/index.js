import Vue from 'vue'
import Vuex from 'vuex'

import catalogModule from './modules/catalog'
import basketModule from './modules/basket'

Vue.use(Vuex)

export function getDefaultStore () {
  return {
    modules: {
      catalog: catalogModule,
      basket: basketModule
    }
  }
}

export default new Vuex.Store(getDefaultStore())
