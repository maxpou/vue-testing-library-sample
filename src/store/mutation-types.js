export const RECEIVE_STARSHIPS = 'RECEIVE_STARSHIPS'
export const INCREMENT_PAGE = 'INCREMENT_PAGE'

export const ADD_TO_BASKET = 'ADD_TO_BASKET'
export const REMOVE_FROM_BASKET = 'REMOVE_FROM_BASKET'
export const INCREMENT_BASKET_ITEM = 'INCREMENT_BASKET_ITEM'
