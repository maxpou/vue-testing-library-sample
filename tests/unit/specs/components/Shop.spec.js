import { mount, shallow } from 'vue-test-utils'
import Selection from '@/components/Shop.vue'
import store from '@/store'

describe('Shop.vue', () => {
  it('test initial rendering', () => {
    const wrapper = mount(Selection, { store })
    const template = wrapper.html()
    expect(template).toMatchSnapshot()
  })

  it('test rendering with datas', () => {
    const wrapper = shallow(Selection, { store })
    const template = wrapper.html()
    expect(template).toMatchSnapshot()
  })
})
