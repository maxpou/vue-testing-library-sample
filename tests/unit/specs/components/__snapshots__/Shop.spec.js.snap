// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop.vue test initial rendering 1`] = `"<div><h2>Shop</h2> <div class=\\"ui container shop\\"><div class=\\"ui cards three column\\"></div></div> <div class=\\"ui container load\\"><!----></div></div>"`;

exports[`Shop.vue test rendering with datas 1`] = `
"<div><h2>Shop</h2> <div class=\\"ui container shop\\"><div class=\\"ui cards three column\\"><div class=\\"card\\"><div class=\\"content\\"><div class=\\"header\\">Executor</div> <div class=\\"meta\\">Kuat Drive Yards, Fondor Shipyards</div> <div class=\\"description\\"><i class=\\"bitcoin icon\\"></i> 1143350000
          </div></div> <div class=\\"ui bottom orange attached button\\"><i class=\\"plus icon\\"></i> add to selection
        </div></div><div class=\\"card\\"><div class=\\"content\\"><div class=\\"header\\">Sentinel-class landing craft</div> <div class=\\"meta\\">Sienar Fleet Systems, Cyngus Spaceworks</div> <div class=\\"description\\"><i class=\\"bitcoin icon\\"></i> 240000
          </div></div> <div class=\\"ui bottom orange attached button\\"><i class=\\"plus icon\\"></i> add to selection
        </div></div><div class=\\"card\\"><div class=\\"content\\"><div class=\\"header\\">Death Star</div> <div class=\\"meta\\">Imperial Department of Military Research, Sienar Fleet Systems</div> <div class=\\"description\\"><i class=\\"bitcoin icon\\"></i> 1000000000000
          </div></div> <div class=\\"ui bottom orange attached button\\"><i class=\\"plus icon\\"></i> add to selection
        </div></div><div class=\\"card\\"><div class=\\"content\\"><div class=\\"header\\">Millennium Falcon</div> <div class=\\"meta\\">Corellian Engineering Corporation</div> <div class=\\"description\\"><i class=\\"bitcoin icon\\"></i> 100000
          </div></div> <div class=\\"ui bottom orange attached button\\"><i class=\\"plus icon\\"></i> add to selection
        </div></div><div class=\\"card\\"><div class=\\"content\\"><div class=\\"header\\">Y-wing</div> <div class=\\"meta\\">Koensayr Manufacturing</div> <div class=\\"description\\"><i class=\\"bitcoin icon\\"></i> 134999
          </div></div> <div class=\\"ui bottom orange attached button\\"><i class=\\"plus icon\\"></i> add to selection
        </div></div><div class=\\"card\\"><div class=\\"content\\"><div class=\\"header\\">X-wing</div> <div class=\\"meta\\">Incom Corporation</div> <div class=\\"description\\"><i class=\\"bitcoin icon\\"></i> 149999
          </div></div> <div class=\\"ui bottom orange attached button\\"><i class=\\"plus icon\\"></i> add to selection
        </div></div><div class=\\"card\\"><div class=\\"content\\"><div class=\\"header\\">TIE Advanced x1</div> <div class=\\"meta\\">Sienar Fleet Systems</div> <!----></div> <div class=\\"ui disabled button\\">Sold out</div></div><div class=\\"card\\"><div class=\\"content\\"><div class=\\"header\\">Slave 1</div> <div class=\\"meta\\">Kuat Systems Engineering</div> <!----></div> <div class=\\"ui disabled button\\">Sold out</div></div><div class=\\"card\\"><div class=\\"content\\"><div class=\\"header\\">Imperial shuttle</div> <div class=\\"meta\\">Sienar Fleet Systems</div> <div class=\\"description\\"><i class=\\"bitcoin icon\\"></i> 240000
          </div></div> <div class=\\"ui bottom orange attached button\\"><i class=\\"plus icon\\"></i> add to selection
        </div></div><div class=\\"card\\"><div class=\\"content\\"><div class=\\"header\\">EF76 Nebulon-B escort frigate</div> <div class=\\"meta\\">Kuat Drive Yards</div> <div class=\\"description\\"><i class=\\"bitcoin icon\\"></i> 8500000
          </div></div> <div class=\\"ui bottom orange attached button\\"><i class=\\"plus icon\\"></i> add to selection
        </div></div></div></div> <div class=\\"ui container load\\"><button class=\\"fluid ui grey basic button\\">
      load more data
    </button></div></div>"
`;
